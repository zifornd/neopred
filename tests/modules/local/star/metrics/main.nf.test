nextflow_process {

    name "Test Process STAR_METRICS"
    script "modules/local/star/metrics/main.nf"
    process "STAR_METRICS"

 test("Should run STAR_METRICS without failures") {

        when {
            process {
                """
                input[0] = [
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/index/star/star.tar.gz', checkIfExists: true)
                ]
                
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.csv},
                { process.out.versions}
            )
        }

    }
    test("Should run GTF2BED with stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/index/star/star.tar.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.csv},
                { process.out.versions},
		{assert snapshot(process.out).match()}
            )
        }

    }

}

