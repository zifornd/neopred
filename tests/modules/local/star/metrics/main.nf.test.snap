{
    "Should run GTF2BED with stub": {
        "content": [
            {
                "0": [
                    "star_metrics.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "versions.yml:md5,cadcd354dd94f5af11f2dde7d062a2f7"
                ],
                "csv": [
                    "star_metrics.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,cadcd354dd94f5af11f2dde7d062a2f7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2025-01-07T07:05:30.877915"
    }
}